pkg/policy/rule.go

```diff
@@ -237,9 +237,11 @@ func (state *traceState) unSelectRule(ctx *SearchContext, labels labels.LabelArr
 
 // resolveL4IngressPolicy determines whether (TODO ianvernon)
 func (r *rule) resolveL4IngressPolicy(ctx *SearchContext, state *traceState, result *L4Policy, requirements []v1.LabelSelectorRequirement) (*L4Policy, error) {
-	if !r.EndpointSelector.Matches(ctx.To) {
-		state.unSelectRule(ctx, ctx.To, r)
-		return nil, nil
+	if !ctx.rulesSelect {
```

Comment:
For what it's worth, somewhat of a nit, but I noticed that `rulesSelect` is only ever checked with `if !ctx.rulesSelect`. This would typically be a bit easier to read if the logic is reversed with a name that allows positive checks. For instance, looking at this 6-line chunk, I read "if not rulesSelect and the rule doesn't match the context", so then in my mind I'm thinking how to reverse this logic to figure out in which cases the function will continue. If this were instead something like `if ctx.checkIfRuleSelectsContext && !r.EndpointSelectorMatches(...)` then I could read "if we need to check that the rule selects the context, and we do so (and it doesn't match), then we need to unselect and return".